saslauthd/auth_krb5.c

```diff
@@ -142,7 +143,15 @@ form_principal_name (
     if ((plen <= 0) || (plen >= pnamelen))
 	return -1;
 
-    /* Perhaps we should uppercase the realm? */
+    /* Uppercase the realm. */
+    if (config && cfile_getswitch(config, "krb5_uppercase_realm", 0)) {
+        if (realm && realm[0]) {
+            char *at;
+            for (at = strrchr(pname, '@'); *at; at++) {
+                *at = toupper(*at);
```

Comment:
Usually code to do this is written `if (islower(c)) c = toupper(c);` I suppose there's nothing exotic in a realm name that will misbehave here though.